exonum/src/storage/db.rs

```diff
@@ -254,66 +260,90 @@ impl Fork {
         if !self.logged {
             panic!("call rollback before checkpoint");
         }
-        for (k, c) in self.changelog.drain(..).rev() {
-            match c {
-                Some(change) => self.changes.insert(k, change),
-                None => self.changes.remove(&k),
-            };
+        for (name, k, c) in self.changelog.drain(..).rev() {
+            if let Some(changes) = self.patch.get_mut(&name) {
+                match c {
+                    Some(change) => changes.insert(k, change),
+                    None => changes.remove(&k),
+                };
+            }
         }
         self.logged = false;
     }
 
-    /// Inserts the key-value pair into the fork.
-    pub fn put(&mut self, key: Vec<u8>, value: Vec<u8>) {
+    /// Inserts the key-value pair into the fork with name `name`.
```

Comment:
...with **_the_** name `name`. :smile: Maybe paraphrase it as "...into the fork with given name"?